concise_concepts/conceptualizer/Conceptualizer.py

```diff
@@ -453,16 +425,16 @@ def assign_score_to_entities(self, doc: Doc):
                 if self.check_presence_vocab(ent.text):
                     entity = [ent.text]
                 else:
-                    entity = [
-                        self.check_presence_vocab(part)
-                        for part in ent.text.split()
-                        if self.check_presence_vocab(part)
-                    ]
-                concept = [
-                    self.check_presence_vocab(word)
-                    for word in self.data_upper[ent.label_]
-                    if self.check_presence_vocab(word)
-                ]
+                    entity = []
+                    for part in ent.text.split():
+                        present_part = self.check_presence_vocab(part)
+                        if present_part:
+                            entity.append(present_part)
+                concept = []
+                for word in self.data_upper[ent.label_]:
+                    present_word = self.check_presence_vocab(word)
+                    if present_word:
+                        concept.append(present_word)
```

Comment:
I've had to break down these list comprehensions to avoid duplicate calls.
I can't wait for ~7 months from now for when Python 3.7 becomes deprecated (https://endoflife.date/python), because then I can freely use the Python 3.8 walrus operators *everywhere*:
```python
concept = [
    present_word
    for word in self.data_upper[ent.label_]
    if (present_word := self.check_presence_vocab(word))
]
```